src/Agent.Worker/TaskCommandExtension.cs

```diff
@@ -618,7 +642,7 @@ public void Execute(IExecutionContext context, Command command)
                 }
             }
 
-            context.TaskVariables.Set(name, data, isSecret);
+            context.TaskVariables.Set(name, data, isSecret, isReadOnly);
```

Comment:
should this have named args as well?

Reply:
To me this is self describing, but we do `isReadOnly: isReadOnly` in the other function above. Let's pick one convention :-).

Reply:
Yeah, I'm good either way, ended up naming the args.

Reply:
I think it was needed elsewhere because of positional args making things harder to read